clio-server/src/main/scala/org/broadinstitute/clio/server/service/SearchService.scala

```diff
@@ -31,8 +30,12 @@ class SearchService private (searchDAO: SearchDAO) {
     transferInput: TI,
     index: ElasticsearchIndex[D],
     queryMapper: ElasticsearchQueryMapper[TI, TO, D]
-  ): Future[Seq[TO]] = {
-    searchDAO.queryMetadata(transferInput, index, queryMapper)
+  )(implicit ec: ExecutionContext): Future[Seq[TO]] = {
+    if (queryMapper.isEmpty(transferInput)) {
+      Future.successful(Seq.empty)
+    } else {
+      searchDAO.queryMetadata(queryMapper.buildQuery(transferInput), index) map queryMapper.toQueryOutputs
```

Comment:
Nitpick: I don't think we use much infix notation, could you instead do `searchDAO.queryMetadata(...).map(queryMapper.toQueryOutputs)`?

Reply:
OK